pkg/query-service/app/metricsexplorer/summary.go

```diff
@@ -447,3 +449,84 @@ func getQueryRangeForRelateMetricsList(metricName string, scores metrics_explore
 
 	return &query
 }
+
+func (receiver *SummaryService) GetInspectMetrics(ctx context.Context, params *metrics_explorer.InspectMetricsRequest) (*metrics_explorer.InspectMetricsResponse, *model.ApiError) {
+	// Capture the original context.
+	parentCtx := ctx
+
+	// Create an errgroup using the original context.
+	g, egCtx := errgroup.WithContext(ctx)
+
+	var attributes []metrics_explorer.Attribute
+	var resourceAttrs map[string]uint64
+
+	// Run the two queries concurrently using the derived context.
+	g.Go(func() error {
+		attrs, apiErr := receiver.reader.GetAttributesForMetricName(egCtx, params.MetricName)
+		if apiErr != nil {
+			return apiErr
+		}
+		if attrs != nil {
+			attributes = *attrs
+		}
+		return nil
+	})
+
+	g.Go(func() error {
+		resAttrs, apiErr := receiver.reader.GetMetricsAllResourceAttributes(egCtx, params.Start, params.End)
+		if apiErr != nil {
+			return apiErr
+		}
+		if resAttrs != nil {
+			resourceAttrs = resAttrs
+		}
+		return nil
+	})
+
+	// Wait for the concurrent operations to complete.
+	if err := g.Wait(); err != nil {
+		return nil, &model.ApiError{Typ: "InternalError", Err: err}
+	}
+
+	// Use the parentCtx (or create a new context from it) for the rest of the calls.
+	if parentCtx.Err() != nil {
+		return nil, &model.ApiError{Typ: "ContextCanceled", Err: parentCtx.Err()}
+	}
+
+	// Build a set of attribute keys for O(1) lookup.
+	attributeKeys := make(map[string]bool)
+	for _, attr := range attributes {
+		attributeKeys[attr.Key] = true
+	}
+
+	// Filter resource attributes that are present in attributes.
+	var validAttrs []string
+	for attrName := range resourceAttrs {
+		if attributeKeys[attrName] {
+			validAttrs = append(validAttrs, attrName)
+		}
+	}
```

Comment:
Since the resourceAttrs have dot in them, this will never pass and validAttrs would always be empty list